src/util.rs

```diff
@@ -12,28 +12,54 @@ pub async fn sleep_random_ms(max_millis: u64) {
     tokio::time::sleep(duration).await;
 }
 
-/// Given a response value from an Oxide API call, returns:
+pub type OxideApiError = oxide_api::Error<oxide_api::types::Error>;
+
+pub fn unwrap_oxide_api_error<T>(
+    result: core::result::Result<oxide_api::ResponseValue<T>, OxideApiError>,
+) -> core::result::Result<(), OxideApiError> {
+    result.map(|_| ())
+}
+
+/// Given an error response from an Oxide API call, returns:
 ///
-/// - `Ok` if the call succeeded.
 /// - `Ok` if the call failed but produced an error response value, irrespective
 ///   of the type of error response.
 /// - `Err` if the call failed without producing an error response value, e.g.
 ///   because the connection to Nexus was interrupted or because a malformed
 ///   response was received.
-pub fn fail_if_no_response<T, U>(
-    result: core::result::Result<
-        oxide_api::ResponseValue<T>,
-        oxide_api::Error<U>,
-    >,
+pub fn fail_if_no_response<U>(
```

Comment:
Looking at how these are now used, I wonder if it makes sense to turn these into associated functions on `AntagonistError` instead of having them be free functions. I think that might flatten out some of the conditionals in the main stress test loop in main.rs:188-197.

Reply:
I don't see how - they don't make sense for the `AnyhowError` variant?

Reply:
I'd have to prototype it, but I think it may make sense for `AntagonistError` to have some additional variants so the checks in the main loop are less "if this is an API error, then if server errors are fatal, then if this is a 500 error, then bail" and more "if this is a 500 error, and server errors are fatal, then bail." But this is a not-totally-insubstantial refactoring and I don't think I'd necessarily hold this PR for it.

Reply:
Have a look at 12d0e52feab50e4c301da17656dce7398ae3df5f, let me know what you think